### ISSUE TITLE
Custom thumbnails for non image files

### ISSUE THREAD

**Issue by IsGarrido** [safrazik/vue-file-agent#66]
Is there any way to define the thumbnail of a file? For example:
- Showing the first page of a PDF
- Grabbing a thumbnail from a remote server.

Thanks in advance

**Comment by safrazik** [comment]
At the moment it’s not possible for file types other than images and videos, but it looks like a great idea. Will consider adding the feature in the future.

I was thinking about showing PDF thumbnails - as I saw that feature somewhere, but I’m afraid the browser support will be limited. But showing thumbnails with the help of a server is a great idea.

**Comment by shafimpk** [comment]
When trying to upload existing images, I tried the following,

```
<script> 
  export default {

created(){

      var data = JSON.parse(this.images)
      for(var i = 0; i < data.length; i++) {       
          var obj ={
            name: data[i].id,
            url: "https://localhost.test/images/properties/1592112694.jpg",
            type:"image",
            ext: "jpg"
          }
        
        this.fileRecords.push(obj)
      }
    }

}

</script>
```


but when I see the file its not displaying anything.. see the image. What is the possible issue here

![image](https://user-images.githubusercontent.com/13327438/84586771-3983ff00-ae2b-11ea-82e6-b1191291586a.png)


waiting for your reply. 

Thanks

**Comment by mondyfy** [comment]
@R-Evolution have you achieved it in any way? I am also having same issue. @safrazik waiting for this.

**Comment by R-Evolution** [comment]
@mondyfy Hi! 

Sorry for the late response. 

I pushed a solution to https://github.com/R-Evolution/vue-file-agent. 

Here are the changes: https://github.com/R-Evolution/vue-file-agent/commit/cf0bd35e55d0caafb7958a4036e8801309b9611c
I also added a modified version of the basic demo:  https://github.com/R-Evolution/vue-file-agent/blob/master/src/demo/demo-basic-custom-thumbnails.vue

Summing up:

1. We will use the filesSelected event to intercept the file before upload begins
`      @select="filesSelected($event)"
`

2. In this method, just set customThumbnail to whatever you want. I used a mock pickture as an example.

```
filesSelected: function(fileRecords) {
      // manual handling
      // this.$refs.vueFileInput.upload(this.uploadUrl, fileRecords);
      fileRecords.forEach(function(val, idx){
        val.customThumbnail = 'https://picsum.photos/200/300.jpg?random=' + Math.random()*10;
      });
      let result = this.$refs.vueFileInput.upload(this.uploadUrl, {}, fileRecords);
      //this.$refs.vueFileInput.upload(this.uploadUrl, {}, fileRecords, function(){});
    },
```
As  @safrazik said it will be interesting having a PDF preview but It has some problemas. I will call a service to create the image and use as described. It is an open aproach that might not suit this project.
